From bbab6920a071d2bda193fbc71df466de8cd43a35 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Wed, 10 Jan 2024 09:32:37 +0530
Subject: [PATCH] Fixed for Crash in com.android.phone during Stability test.

Since we are not supporting telephony module in AAOS, So check for it
before fetching cell location.

Tests: Run stability test and no crash has observed.

Tracked-On: OAM-113415
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 telephony/java/android/telephony/TelephonyManager.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/telephony/java/android/telephony/TelephonyManager.java b/telephony/java/android/telephony/TelephonyManager.java
index 8bc4d381e2dd..1ffe4e532caf 100644
--- a/telephony/java/android/telephony/TelephonyManager.java
+++ b/telephony/java/android/telephony/TelephonyManager.java
@@ -49,6 +49,7 @@ import android.compat.annotation.UnsupportedAppUsage;
 import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
+import android.content.pm.PackageManager;
 import android.database.Cursor;
 import android.net.ConnectivityManager;
 import android.net.Uri;
@@ -2348,6 +2349,9 @@ public class TelephonyManager {
     @Deprecated
     @RequiresPermission(android.Manifest.permission.ACCESS_FINE_LOCATION)
     public CellLocation getCellLocation() {
+        if (!mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEPHONY)) {
+            return null;
+        }
         try {
             ITelephony telephony = getITelephony();
             if (telephony == null) {
-- 
2.17.1

